File: src/main/java/com/back/domain/auction/auction/entity/AuctionImage.java
package com.back.domain.auction.auction.entity;

import com.back.domain.image.image.entity.Image;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "auction_images")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@IdClass(AuctionImageId.class)
public class AuctionImage {

    @Id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "auction_id")
    private Auction auction;

    @Id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "image_id")
    private Image image;

    public AuctionImage(Auction auction, Image image) {
        this.auction = auction;
        this.image = image;
    }

    public void setAuction(Auction auction) {
        this.auction = auction;
    }
}

File: src/main/java/com/back/domain/auction/auction/entity/AuctionImage.kt
package com.back.domain.auction.auction.entity

import com.back.domain.image.image.entity.Image
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.IdClass
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "auction_images")
@IdClass(AuctionImageId::class)
class AuctionImage() {
    @Id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "auction_id")
    lateinit var auction: Auction

    @Id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "image_id")
    lateinit var image: Image

    constructor(auction: Auction, image: Image) : this() {
        this.auction = auction
        this.image = image
    }

    fun setAuction(auction: Auction) {
        this.auction = auction
    }
}
